src/components/Notification/Notification.tsx

```diff
@@ -198,15 +199,21 @@ const Notification = ({
           {hasActions ? (
             <div className="p-notification__actions">
               {actions.map((action, i) => (
-                <Button
-                  appearance={ButtonAppearance.LINK}
-                  className="p-notification__action"
-                  data-testid="notification-action"
-                  key={`${action.label}-${i}`}
-                  onClick={action.onClick}
-                >
-                  {action.label}
-                </Button>
+                <Fragment key={i}>
+                  {typeof action === "object" && "label" in action ? (
```

Comment:
This will also need to check that `action` exists, as `typeof null === "object"` but would then fail when it tried to check `"label" in null`.
```suggestion
                  {action && typeof action === "object" && "label" in action ? (
```

Reply:
Can you take another look? I had to use `isValidElement` to make it work